packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -458,6 +461,28 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         return this.channel.getGCData(fullGC);
     }
 
+    /**
+     * After GC has run, called to notify this data stores of routes whose unreferenced state needs to be reset. These
+     * are routes to nodes in the data store that have transitioned from `unreferenced -> referenced -> unreferenced`
+     * since the last GC run.
+     * @param routesToReset - The routes to reset in this data store.
+     */
+    public resetUnreferencedState(routesToReset: string[]) {
```

Comment:
I wonder if it should be responsibility of "central GC mechanism" to track all the state in summaries (like last referenced state) vs. push this responsibility to data stores (if that is possible, of course).
I.e. ideally data it's not hard to implement data store contract. And if each data store needs to do the same thing, then we either need to consider moving it out and parent later (container runtime doing it), or provide stock implementation as a building block (hard in this case due to nesting).
Basically - instead of data stores having .gc nodes in summaries, can container runtime summarize and manage all that data in parallel summary tree?